app/scripts/com/2fdevs/videogular/directives/vg-media.js

```diff
@@ -43,31 +44,51 @@ angular.module("com.2fdevs.videogular")
 
                 // FUNCTIONS
                 scope.onChangeSource = function onChangeSource(newValue, oldValue) {
-                    if ((!sources || newValue != oldValue) && newValue) {
+                    if (newValue && (!sources || JSON.stringify(newValue) != JSON.stringify(oldValue)) && sources !== newValue) {
```

Comment:
For the source change test here, it might be better to use angular.equals(). Any other thoughts on the logic?